services/npm/npm-last-update.service.js

```diff
@@ -0,0 +1,100 @@
+import Joi from 'joi'
+import dayjs from 'dayjs'
+import { NotFound, pathParam, queryParam } from '../index.js'
+import { formatDate } from '../text-formatters.js'
+import { age as ageColor } from '../color-formatters.js'
+import NpmBase, { packageNameDescription } from './npm-base.js'
+
+const updateResponseSchema = Joi.object({
+  time: Joi.object({
+    created: Joi.string().required(),
+    modified: Joi.string().required(),
+  })
+    .pattern(Joi.string(), Joi.any())
+    .required(),
+  'dist-tags': Joi.object().pattern(Joi.string(), Joi.string()).required(),
+}).required()
```

Comment:
This schema could do with tightening up a bit. My suggestion would be:

```suggestion
const updateResponseSchema = Joi.object({
  time: Joi.object({
    created: Joi.string().required(),
    modified: Joi.string().required(),
  })
    .pattern(Joi.string().required(), Joi.string().required())
    .required(),
  'dist-tags': Joi.object()
    .pattern(Joi.string().required(), Joi.string().required())
    .required(),
}).required()
```

@jnullj As a tip for future reviews:

It is not invalid to ever use `Joi.any()` in a schema, but it is quite rare that we want to be that loose. Usually if you see that in a PR, the next question is: "can we tighten this up?" Similarly, it isn't invalid for something in a schema to ever be optional, but usually any value we're going to use should be required.

If the upstream API does require us to accept optional values or variable types in the schema, then we generally need some additional checks in the handler code to compensate for it.